### PR TITLE
fix/step3

### PR DESCRIPTION
**Pull request by chancyk**
OS: Windows 10
Nim: 1.6.2

I was getting the following two errors for Step 3, which I was able to resolve with these changes.

* `ortho` seemed to be creating a GMat4[system.int] given the literals, so I just did `ortho[float32]()` instead.
* `addr mvp[0]` seemed to be confused getting a Vec4 instead of an `array`. I enabled the `-d:vmathArrayBased` define and then casted it to a `ptr GLFloat` which seems to solve the problem?

I'm out of my element but this worked for me!

### Error 1
```bash
step3.nim(91, 12) Error: type mismatch:
got 'GMat4[system.int]' for 'ortho(-1, 1, 1, -1, -1000, 1000)' but
expected 'Mat4 = GMat4[system.float32]'
```

### Error 2
```bash
step3.nim(95, 62) Error: expression has no address; maybe use 'unsafeAddr'
```

### CODE REVIEW

File: step3.nims [hunk 1]
```diff
@@ -21,6 +21,7 @@ if defined(emscripten):
   --gc:arc # GC:arc is friendlier with crazy platforms.
   --exceptions:goto # Goto exceptions are friendlier with crazy platforms.
   --define:noSignalHandler # Emscripten doesn't support signal handlers.
+  --define:vmathArrayBased # For interop with OpenGL
```
Comment (treeform) [inline step3.nims]: `--define:vmathArrayBased` this is not a good idea array based is slower. ObjArray default is like 20% faster. There should be other ways to cast that.

File: step3.nim [hunk 2]
```diff
@@ -88,11 +88,12 @@ proc mainLoop() {.cdecl.} =
   glClear(GL_COLOR_BUFFER_BIT)
 
   m = rotateZ(getTime().float32)
-  p = ortho(-1, 1, 1, -1, -1000, 1000)
+  p = ortho[float32](-1, 1, 1, -1, -1000, 1000)
   mvp = m * p;
 
   glUseProgram(program)
-  glUniformMatrix4fv(mvpLocation.GLint, 1, GL_FALSE, addr mvp[0]);
+  var mvp_addr = cast[ptr GLFloat](addr mvp[0])
```
Comment (treeform) [inline step3.nim]: if you don't use `--define:vmathArrayBased` does `cast[ptr GLFloat](mvp.unsafeAddr)` work?

**Comment by treeform** [comment]
Thank you for the PR!

File: step3.nim (hunk 2 above)
Comment (chancyk) [reply step3.nim]: That did work actually! I made the change to unsafeAddr and removed the --define from `step3.nims`.

**Comment by treeform** [comment]
Great PR thank you for catching the issue.

**Pull request merged**